client.go

```diff
@@ -5,11 +5,32 @@ import (
 	"io/ioutil"
 	"net/url"
 
+	"github.com/pkg/errors"
+
 	"github.com/hashicorp/vault/api"
 )
 
 type Client struct {
 	*api.Client
+
+	auth AuthProvider
+}
+
+type Service struct {
+	client     *Client
+	MountPoint string
+}
+
+type RequestOptions struct {
+	Parameters url.Values
+
+	// RenewToken defines if the client should retry this Request with a new Token if it fails because of
+	// 403 Permission Denied
+	// The default behavior of the client is to always Request a new Token on 403
+	// Only if this is explicitly set to false, the client will continue processing the first failed request
+	// This should generally only be disabled for TokenAuth requests (a failed TokenAuth request can't be fixed by
+	// doing another TokenAuth request, this would lead to infinite recursion)
+	RenewToken *bool
```

Comment:
Might it be easier to turn this around and introduce a `SkipRenewal` parameter that has a null value of `false` and doesn't need to be a pointer?

Reply:
This does indeed simplify the code a lot. I didn't want to introduce a double negative like `DisableRenewal`, but `SkipRenewal` sounds good.